tools/iree-benchmark-executable-main.c

```diff
@@ -0,0 +1,577 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "iree/base/api.h"
+#include "iree/base/internal/file_io.h"
+#include "iree/base/internal/flags.h"
+#include "iree/hal/api.h"
+#include "iree/modules/hal/types.h"
+#include "iree/testing/benchmark.h"
+#include "iree/tooling/device_util.h"
+#include "iree/tooling/function_io.h"
+#include "iree/vm/api.h"
+
+IREE_FLAG(
+    int32_t, batch_size, 64,
+    "Number of dispatches to perform per command buffer submission.\n"
+    "Higher numbers will reduce the effect of submission overheads on the\n"
+    "final timings but too high a value may result in hangs.");
+
+IREE_FLAG(string, executable_format, "",
+          "Format of the executable file being loaded.");
+IREE_FLAG(string, executable_file, "", "Path to the executable file to load.");
+
+IREE_FLAG(int32_t, entry_point, 0, "Entry point ordinal to run.");
+
+IREE_FLAG_LIST(
+    string, workgroup_count,
+    "`x,y,z` dimensions of the workgroup count defining the number of\n"
+    "workgroup invocations that will be run per benchmark iteration.\n"
+    "Each occurrence of the flag will run a benchmark with that set of\n"
+    "workgroup count values.");
+
+// Total number of executable-level constants we (currently); this is only a
```

Comment:
```suggestion
// Total number of executable-level constants we (currently) allow. This is only a
```

Also what is an executable-level constant?

Reply:
aka specialization constants in SPIR-V or device constants (symbols) in CUDA. For us they are set once at executable creation time on the host and then accessible to the HAL executable loader (for specializing/JITing/etc) and within the dispatches as values. They can be used for things like local workgroup size, device feature support, etc that can be queried on the host and passed in. They're what we should be using instead of hardcoding attributes in the compiler.

Reply:
Ah ok, I guess that's why I never see them :P